pybop/models/lithium_ion/basic_SPMe.py

```diff
@@ -0,0 +1,619 @@
+            v_s_n = pybamm.x_average(eta_n + U_n)
+            v_s_p = pybamm.x_average(eta_p + U_p)
+
+        ######################
+        # Double layer
+        ######################
+        if include_double_layer:
+            self.rhs[v_s_n] = 1 / C_n * (I - 3 * Q_th_n * pybamm.x_average(j_n))
+            self.rhs[v_s_p] = 1 / C_p * (-I - 3 * Q_th_p * pybamm.x_average(j_p))
+
+        sto_n_init = x_0 + (x_100 - x_0) * soc_init
+        sto_p_init = y_0 + (y_100 - y_0) * soc_init
+        U_n_init = self.U(sto_n_init, "negative")
+        U_p_init = self.U(sto_p_init, "positive")
+
+        if include_double_layer:
+            self.initial_conditions[v_s_n] = U_n_init
+            self.initial_conditions[v_s_p] = U_p_init
+
+        ######################
+        # Particles
+        ######################
+        # The div and grad operators will be converted to the appropriate matrix
+        # multiplication at the discretisation stage
+        self.rhs[sto_n] = pybamm.div(pybamm.grad(sto_n) / tau_d_n)
+        self.rhs[sto_p] = pybamm.div(pybamm.grad(sto_p) / tau_d_p)
+
+        # Boundary conditions must be provided for equations with spatial derivatives
+        self.boundary_conditions[sto_n] = {
+            "left": (Scalar(0), "Neumann"),
+            "right": (-tau_d_n * pybamm.x_average(j_n), "Neumann"),
+        }
+        self.boundary_conditions[sto_p] = {
+            "left": (Scalar(0), "Neumann"),
+            "right": (-tau_d_p * pybamm.x_average(j_p), "Neumann"),
+        }
+
+        self.initial_conditions[sto_n] = sto_n_init
+        self.initial_conditions[sto_p] = sto_p_init
+
+        ######################
+        # Electrolyte
+        ######################
+        self.rhs[sto_e_n] = (
+            pybamm.div(pybamm.grad(sto_e_n) / tau_e_n - (t_plus * I / Q_e) * x_n / l_n)
+            + (3 / Q_e) * Q_th_n * j_n / l_n
+        ) / zeta_n
+        self.rhs[sto_e_sep] = pybamm.div(
+            pybamm.grad(sto_e_sep) / tau_e_sep - t_plus * I / Q_e
+        )
+        self.rhs[sto_e_p] = (
+            pybamm.div(
+                pybamm.grad(sto_e_p) / tau_e_p - (t_plus * I / Q_e) * (1 - x_p) / l_p
+            )
+            + (3 / Q_e) * Q_th_p * j_p / l_p
+        ) / zeta_p
+
+        self.boundary_conditions[sto_e_n] = {
+            "left": (Scalar(0), "Neumann"),
+            "right": (
+                tau_e_n * pybamm.boundary_gradient(sto_e_sep, "left") / tau_e_sep,
+                "Neumann",
+            ),
+        }
+        self.boundary_conditions[sto_e_sep] = {
+            "left": (pybamm.boundary_value(sto_e_n, "right"), "Dirichlet"),
+            "right": (pybamm.boundary_value(sto_e_p, "left"), "Dirichlet"),
+        }
+        self.boundary_conditions[sto_e_p] = {
+            "left": (
+                tau_e_p * pybamm.boundary_gradient(sto_e_sep, "right") / tau_e_sep,
+                "Neumann",
+            ),
+            "right": (Scalar(0), "Neumann"),
+        }
```

Comment:
If variables are concatenated into `sto_e`, then we should be able to define an equation for `sto_e` and PyBaMM would deal with the continuity conditions.